be/src/runtime/primitive_type.h

```diff
@@ -81,6 +82,10 @@ inline bool is_enumeration_type(PrimitiveType type) {
 }
 
 inline bool is_type_compatible(PrimitiveType lhs, PrimitiveType rhs) {
+    if (lhs == TYPE_FUNCTION || rhs == TYPE_FUNCTION) {
```

Comment:
What if lhs and rhs are both TYPE_FUNCTION?

Reply:
Function data type shouldn't be operators for any operations, so making it invalid in any case.